core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -718,6 +738,28 @@ else if (!found && result) {
             return false;
         }
 
+        private List<Object> processInListExpression(Object value, InListExpression valueList, Object context)
```

Comment:
Why is this together with the introduction of `optimizationCache` cache?
It looks like the introduction of `optimizationCache` should induce changes in `process` method only.